### PR TITLE
Changed Dockerfile for USER/ID/GID to be Docker build arguments

### PR DESCRIPTION
**Pull request by rwmajor2**
#719

### CODE REVIEW

File: base-notebook/Dockerfile [hunk 1]
```diff
@@ -6,6 +6,9 @@
 FROM ubuntu:bionic-20180526@sha256:c8c275751219dadad8fa56b3ac41ca6cb22219ff117ca98fe82b42f24e1ba64e
 
 LABEL maintainer="Jupyter Project <jupyter@googlegroups.com>"
+ARG BUILD_USER="jovyan"
+ARG BUILD_UID="1000"
+ARG BUILD_GID="100"
```
Comment (parente) [inline base-notebook/Dockerfile]: Maybe I'm mistaken, but I think the ARG names can match the ENV names.

**Comment by rwmajor2** [comment]
@parente Yes, you are correct.  I have updated the Dockerfile and tested to ensure it works.  Please review.

**Comment by parente** [comment]
The build failure for this PR has happened twice in a row now, but what's causing it is not jumping out at me in the log: https://travis-ci.org/jupyter/docker-stacks/builds/432264734. In fact, the last lines say:

```
The command "make build-test-all DARGS="--build-arg TEST_ONLY_BUILD=1"" exited with 0.

Done. Your build exited with 1.
```

Huh?

In contrast, the daily rebuild of master triggered by travis cron has been succeeding: https://travis-ci.org/jupyter/docker-stacks/builds/433705194.

**Comment by parente** [comment]
@rwmajor2 sure can you rebase the PR on master?

**Comment by rwmajor2** [comment]
@parente Done.

**Comment by parente** [comment]
LGTM now. Let's ship it. Thanks @rwmajor2. 🍰

**Pull request merged**